test/unit/platform_triple/platform_triple_test.bzl

```diff
@@ -0,0 +1,109 @@
+"""Tests for the platform triple constructor"""
+
+load("@bazel_skylib//lib:unittest.bzl", "asserts", "unittest")
+load("//rust/platform:triple.bzl", "triple")
+load("//rust/platform:triple_mappings.bzl", "SUPPORTED_PLATFORM_TRIPLES")
+
+def _construct_platform_triple_test_impl(ctx):
+    env = unittest.begin(ctx)
+
+    imaginary_triple = triple("arch-vendor-system-abi")
+
+    asserts.equals(
+        env,
+        "arch",
+        imaginary_triple.arch,
+    )
+
+    asserts.equals(
+        env,
+        "vendor",
+        imaginary_triple.vendor,
+    )
+
+    asserts.equals(
+        env,
+        "system",
+        imaginary_triple.system,
+    )
+
+    asserts.equals(
+        env,
+        "abi",
+        imaginary_triple.abi,
+    )
+
+    asserts.equals(
+        env,
+        "arch-vendor-system-abi",
+        imaginary_triple.str,
+    )
+
+    return unittest.end(env)
+
+def _construct_minimal_platform_triple_test_impl(ctx):
+    env = unittest.begin(ctx)
+
+    imaginary_triple = triple("arch-vendor-system")
+
+    asserts.equals(
+        env,
+        "arch",
+        imaginary_triple.arch,
+    )
+
+    asserts.equals(
+        env,
+        "vendor",
+        imaginary_triple.vendor,
+    )
+
+    asserts.equals(
+        env,
+        "system",
+        imaginary_triple.system,
+    )
+
+    asserts.equals(
+        env,
+        "arch-vendor-system",
+        imaginary_triple.str,
+    )
```

Comment:
maybe add an assert for how .abi behaves when it's missing

Reply:
Great idea, done!